arduino-core/src/cc/arduino/contributions/DownloadableContributionsDownloader.java

```diff
@@ -199,11 +200,9 @@ public boolean verifyDomain(URL url) {
   public boolean checkSignature(MultiStepProgress progress, URL signatureUrl, ProgressListener progressListener, SignatureVerifier signatureVerifier, String statusText, File fileToVerify) throws Exception {
 
     final boolean allowInsecurePackages =
-      PreferencesData.getBoolean("allow_insecure_packages", false);
-    if (allowInsecurePackages) {
-      log.info("Allow insecure packages is true the signature will be skip and return always verified");
-      return true;
-    }
+      PreferencesData.getBoolean(Constants.ALLOW_INSECURE_PACKAGES, false);
+    final boolean trustAll = PreferencesData.getBoolean(Constants.PREF_CONTRIBUTIONS_TRUST_ALL);
+    final boolean skipVerification = allowInsecurePackages || trustAll;
```

Comment:
What about adding a method `PreferencesData.areInsecurePackagesAllowed()` and call that method every time we need to check these preferences? something along the line of:

```
public static boolean areInsecurePackagesAllowed() {
  if (getBoolean("allow_insecure_packages", false)) {
    log.info("allow_insecure_packages preference is true, the signature check will be ignored.");
    return true;
  }
  if (getBoolean("contributions.trust.all", false)) {
    log.info("contributions.trust.all preference is true, the signature check will be ignored.");
    return true;
  }
  return false;
}
```
We could remove also the two `Constants.ALLOW_INSECURE_PACKAGES` and `Constants.PREF_CONTRIBUTIONS_TRUST_ALL`.

Reply:
Done https://github.com/arduino/Arduino/pull/9096/commits/e0374c115a5e842e4f2d2f4438676b4679ed15e5